### PR TITLE
Update EIP-6381 interface

### PR DESCRIPTION
**Pull request by ThunderDeliverer**
# Description

The EIP-6381 interface was updated to include the latest version of the proposal.
There was a misidentification where IRMRKEmotes was labeled as IERC6381, where IRMRKEmoteTracker was the one that should have been renamed. This is now fixed, and the appropriate interfaces are assigned to the appropriate extensions.

# Checklist

- [x] Verified code additions
- [x] Updated NatSpec comments (if applicable)
- [x] Regenerated docs
- [x] Ran prettier
- [ ] Added tests to fully cover the changes

---

## PR-Codex overview
This PR removes the `IRMRKEmoteTracker` interface and replaces it with `IRMRKEmotable` in the `RMRKEmotable` contract. It also adds new functions to the `RMRKEmoteTrackerMock` contract.

### Detailed summary
- Remove `IRMRKEmoteTracker` interface and replace with `IRMRKEmotable` in `RMRKEmotable` contract
- Add new functions to `RMRKEmoteTrackerMock` contract:
  - `bulkEmote`
  - `bulkEmoteCountOf`
  - `bulkPrepareMessagesToPresignEmote`

> The following files were skipped due to too many changes: `docs/mocks/extensions/emotable/RMRKEmoteTrackerMock.md`, `contracts/RMRK/extension/emotable/IERC6381.sol`, `docs/RMRK/extension/emotable/RMRKEmoteTracker.md`, `contracts/RMRK/extension/emotable/RMRKEmoteTracker.sol`, `docs/RMRK/extension/emotable/IERC6381.md`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by ThunderDeliverer** [comment]
This still needs the tests to be migrated from our proposal, but it serves its purpose for now. I need this as I want to start recalculation of the repository address.

File: contracts/RMRK/extension/emotable/RMRKEmotable.sol [hunk 1]
```diff
@@ -2,21 +2,21 @@
...
-import "./IERC6381.sol";
+import "./IRMRKEmotable.sol";
 
 /**
  * @title RMRKEmotable
  * @author RMRK team
  * @notice Smart contract of the RMRK Emotable module.
  */
-abstract contract RMRKEmotable is IERC6381 {
+abstract contract RMRKEmotable is IRMRKEmotable {
```
[the remark this answers is not in the document]
Comment (ThunderDeliverer) [reply contracts/RMRK/extension/emotable/RMRKEmotable.sol:12]: Emotable is per-collection. So internal to the collection. We either have to rework it to support all of the ERC-6381 methods or address it some other way..

---

I just noticed I never submitted the response 🤦

**Comment by ThunderDeliverer** [comment]
We could modify the extension version to use the repository out of the gate (since the address is predetermined) 🤔

**Comment by steven2308** [comment]
> We could modify the extension version to use the repository out of the gate (since the address is predetermined) 🤔

What do we get from it? I don't see any value, just more code to maintain.

**Pull request merged**